change to client_id and client_secret naming

smrutherford opened the pull request:
Addresses open issue #86 Credentials naming should be consistent - adds deprecation warning for September 2019 and migrates to `client_id` and `client_secret` naming.

Gerdie commented on nylas/client/client.py:
```diff
@@ -89,30 +91,54 @@ def __init__(
+    @property
+    def app_id(self):
+        warn(DeprecationWarning('"app_id" naming convention will be deprecated Sept 2019. '
+                                'Use "client_id" instead.'))
```
np: let's say `app_id` will be deprecated in version 5.0.0. Use `client_id` instead.

smrutherford replied:
Good idea!

smrutherford commented:
> looking good! can we add a test to make sure the warnings are emitting as expected, and a backwards compatibility test to make sure initializing the client with the `app_id` and `client_id` works as expected?

yes! Added a test. However, the warnings don't work as we thought (well, at least what I thought). The warnings only emit if the `-Wd` flag is used when opening python. Reason behind this being that warnings shouldn't emit for the user, but for the developer. So the warnings are shown in the code. Also, I changed from `DeprecationWarning` to `PendingDeprecationWarning` since it's not fully deprecated yet.
